components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -395,12 +395,35 @@ func virtualServiceName(kfName string, namespace string) string {
 	return fmt.Sprintf("notebook-%s-%s", namespace, kfName)
 }
 
+// Function for the VirtualService header blacklist
+func Find(slice []string, val string) (int, bool) {
+	for i, item := range slice {
+		if item == val {
+			return i, true
+		}
+	}
+	return -1, false
+}
+
 func generateVirtualService(instance *v1beta1.Notebook) (*unstructured.Unstructured, error) {
 	name := instance.Name
 	namespace := instance.Namespace
 	clusterDomain := "cluster.local"
 	prefix := fmt.Sprintf("/notebook/%s/%s/", namespace, name)
-	rewrite := fmt.Sprintf("/notebook/%s/%s/", namespace, name)
+	var rewrite string
+	// Create a map of all the annotations and assign to a variable
+	annotations := make(map[string]string)
+	for k, v := range instance.ObjectMeta.Annotations {
+		annotations[k] = v
+	}
+
+	rewrite = fmt.Sprintf("/notebook/%s/%s/", namespace, name)
+	// If key "notebooks.kubeflow.org/http-rewrite-uri" is present in the annotations
+	// set the rewrite variable to the value of base-url
+	if _, ok := annotations["notebooks.kubeflow.org/http-rewrite-uri"]; ok && len(annotations["notebooks.kubeflow.org/http-rewrite-uri"]) > 0 {
```

Comment:
@DavidSpek these values are used a few times:
"notebooks.kubeflow.org/http-rewrite-uri"
"notebooks.kubeflow.org/http-headers-request-set"

Can you make them into constants, so that they're declared in one place?

Reply:
Makes sense.